UM/Qt/qml/UM/MessageStack.qml

```diff
@@ -91,16 +92,122 @@ ListView
                 margins: UM.Theme.getSize("default_margin").width
             }
 
-            height: childrenRect.height
+            height: Math.max(messageTypeIcon.height, messageTitle.height)
+            Item
+            {
+                id: messageTypeIcon
+                visible: messageIcon.source != ""
+                height: UM.Theme.getSize("small_button_icon").height
+                width: visible ? UM.Theme.getSize("small_button_icon").height : 0
+                anchors.verticalCenter: parent.verticalCenter
+                Rectangle
+                {
+                    id: messageIconBackground
+                    height: parent.height
+                    width: parent.width
+                    radius: Math.round(width / 2)
+                }
+                UM.RecolorImage
+                {
+                    id: messageIcon
+                    height: parent.height
+                    width: parent.width
+                    sourceSize.width: width
+                    sourceSize.height: height
+                }
+
+                states:
+                [
+                    State
+                    {
+                        name: "positive"
+                        when: model.message_type == 0
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: UM.Theme.getIcon("Check", "low")
+                            color: UM.Theme.getColor("message_success_icon")
+                        }
+                        PropertyChanges
+                        {
+                            target: messageIconBackground
+                            color: UM.Theme.getColor("message_success_icon_background")
+                        }
+                    },
+                    State
+                    {
+                        name: "neutral"
+                        when: model.message_type == 1
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: ""
+                        }
+                        PropertyChanges
+                        {
+                            target: messageIconBackground
+                            color: "transparent"
+                        }
+                    },
+                    State
+                    {
+                        name: "warning"
+                        when: model.message_type == 2
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: UM.Theme.getIcon("Warning", "low")
+                            color: UM.Theme.getColor("message_warning_icon")
```

Comment:
We already have a "warning" colour. Should we not enforce that colour to be used for all warnings, so that the warning colours are consistent everywhere?
Rather than hard-coding specific theme entries for specific details in the interface, the design philosophy is to have re-usable theme entries as much as possible.

Reply:
I missed the already existing warning color. It is now using that instead of defining its own.